urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlib/classes/SimplePermClass.java

```diff
@@ -0,0 +1,559 @@
+                        DecoratedSimplePerm s = new DecoratedSimplePerm(p, p.length());
+                        DFSProcessSimples(s, k, n, processor);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Processes the simples up to length
+     * <code>n</code> in the class with the given processor.
+     *
+     * @param n the length
+     * @param processor the processor
+     */
+    public void processWithTo(PermProcessor processor, int n) {
+        for (DecoratedSimplePerm s : simples[4]) {
+            DFSProcessSimplesTo(s, 4, n, processor);
+        }
+        for (int k = 6; k <= n; k += 2) {
+            for (Permutation p : exceptionalSimples(k)) {
+                if (inClass.isSatisfiedBy(p)) {
+                    DecoratedSimplePerm s = new DecoratedSimplePerm(p, p.length());
+                    DFSProcessSimplesTo(s, k, n, processor);
+                }
+            }
+        }
+    }
+
+    public void DFSProcessSimples(DecoratedSimplePerm s, int i, int n, PermProcessor processor) {
+        if (i > n) {
+            return;
+        }
+        if (i == n) {
+            processor.process(s.getPermutation());
+            return;
+        }
+        for (DecoratedSimplePerm q : s.leftSimpleExtensions()) {
+            if (inClass.isSatisfiedBy(q.getPermutation())) {
+                DFSProcessSimples(q, i + 1, n, processor);
+            }
+        }
+    }
+
+    private void DFSProcessSimplesTo(DecoratedSimplePerm s, int i, int n, PermProcessor processor) {
+        if (i > n) {
+            return;
+        }
+        processor.process(s.getPermutation());
+        if (i == n) {
+            return;
+        }
+        for (DecoratedSimplePerm q : s.leftSimpleExtensions()) {
+            if (inClass.isSatisfiedBy(q.getPermutation())) {
+                DFSProcessSimplesTo(q, i + 1, n, processor);
+            }
+        }
+    }
+
+    @Override
+    public boolean containsPermutation(Permutation q) {
+        return inClass.isSatisfiedBy(q) && simplicity.isSatisfiedBy(q);
+    }
+
+    public class DecoratedSimplePerm {
+
+        Permutation p;
+        int leftmostSimpleDeletionIndex;
+
+        public DecoratedSimplePerm(Permutation p) {
+            this.p = p;
+            leftmostSimpleDeletionIndex = 0;
+            while (leftmostSimpleDeletionIndex < p.length() && !simplicity.isSatisfiedBy(delete(p, leftmostSimpleDeletionIndex))) {
+                leftmostSimpleDeletionIndex++;
+            }
+        }
+
+        public DecoratedSimplePerm(Permutation p, int leftmostSimpleDeletionIndex) {
+            this.p = p;
+            this.leftmostSimpleDeletionIndex = leftmostSimpleDeletionIndex;
+        }
+
+        public int getLeftmostSimpleDeletionIndex() {
+            return leftmostSimpleDeletionIndex;
+        }
+
+        public Permutation getPermutation() {
+            return p;
+        }
+
+        public ArrayList<DecoratedSimplePerm> leftSimpleExtensions() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 336-364, lines 365-384 into separate methods.